Src/Common/WebHeaderCollectionExtensions.cs

```diff
@@ -124,6 +126,67 @@ public static IEnumerable<string> GetHeaderValue(this NameValueCollection header
             return null;
         }
 
+        public static void ReadActivityBaggage(this NameValueCollection headers, Activity activity)
+        {
+            Debug.Assert(headers != null, "Headers must not be null");
+            Debug.Assert(activity != null, "Activity must not be null");
+
```

Comment:
Do we need to check !activity.Baggage.Any()
https://github.com/microsoft/ApplicationInsights-aspnetcore/blob/develop/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs#L621

Reply:
sure we can in debug.assert, but we call this method from the places we control and there we make sure there is no baggage